app_feup/lib/view/Widgets/bug_report_form.dart

```diff
@@ -169,83 +186,113 @@ class BugReportFormState extends State<BugReportForm> {
 
+    final String title = titleController.text;
     final String bugLabel = bugDescriptions[_selectedBug] == null
         ? 'Unidentified bug'
         : bugDescriptions[_selectedBug].item2;
-    final Map data = {
-      'headline6': titleController.text,
-      'body': descriptionController.text,
-      'labels': [_issueLabel, bugLabel]
-    };
-
-    final url = _postUrl + '?access_token=' + ghToken;
-    http
-        .post(url.toUri(),
-            headers: {'Content-Type': 'application/json'},
-            body: json.encode(data))
-        .then((http.Response response) {
-      final int statusCode = response.statusCode;
-
-      String msg;
-      if (statusCode < 200 || statusCode > 400) {
-        Logger().e('Error ' + statusCode.toString() + ' while posting bug');
-        msg = 'Ocorreu um erro no envio';
-      } else {
-        Logger().i('Successfully submitted bug report.');
-        msg = 'Enviado com sucesso';
-
-        clearForm();
-
-        Navigator.pop(context);
-        setState(() {
-          _isButtonTapped = false;
-        });
-      }
-      
-
-      FocusScope.of(context).requestFocus( FocusNode());
-      displayBugToast(msg);
-      setState(() {
-        _isButtonTapped = false;
-      });
-    }).catchError((error) {
-      Logger().e(error);
-      FocusScope.of(context).requestFocus( FocusNode());
-
-      final String msg =
-          (error is SocketException) ? 'Falha de rede' : 'Ocorreu um erro';
-      displayBugToast(msg);
-      setState(() {
-        _isButtonTapped = false;
+
+    String toastMsg;
+    try {
+      final String sentryDescription = emailController.text == ''
+          ? descriptionController.text
+          : descriptionController.text + '\nContact: ' + emailController.text;
+      final sentryId = await Sentry.captureMessage(
+          bugLabel + ': ' + title + '\n' + sentryDescription);
+
+      final String gitHubDescription =
+          descriptionController.text + '\n' + _sentryLink + sentryId.toString();
+      final Map gitHubData = {
+        'title': titleController.text,
+        'body': gitHubDescription,
+        'labels': [_issueLabel, bugLabel],
+      };
+      http
+          .post(Uri.parse(_gitHubPostUrl),
+              headers: {
+                'Content-Type': 'application/json',
+                'Authorization': 'token $ghToken'
+              },
+              body: json.encode(gitHubData))
+          .then((http.Response response) {
+        final int statusCode = response.statusCode;
+        if (statusCode < 200 || statusCode > 400) {
+          throw Exception('Network error');
+        }
       });
+      Logger().i('Successfully submitted bug report.');
+      toastMsg = 'Enviado com sucesso';
+    } catch (e) {
+      Logger().e('Error while posting bug report:' + e.toString());
+      toastMsg = 'Ocorreu um erro no envio';
+    }
+
+    clearForm();
+    FocusScope.of(context).requestFocus(FocusNode());
+    displayToastMessage(toastMsg);
```

Comment:
Does the toast have error levels to portray the information differently?

Reply:
I think a background color change should do

Reply:
Agreed!